mmv1/third_party/terraform/utils/compute_instance_helpers.go.erb

```diff
@@ -408,5 +417,28 @@ func schedulingHasChange(d *schema.ResourceData) bool {
 		return true
 	}
 
-	return reflect.DeepEqual(newNa, originalNa)
+	return false
+}
+
+func hasNodeAffinitiesChanged(oScheduling, newScheduling map[string]interface{}) bool {
+	oldNAs := oScheduling["node_affinities"].(*schema.Set).List()
+	newNAs := newScheduling["node_affinities"].(*schema.Set).List()
+	if len(oldNAs) != len(newNAs) {
+		return true
+	}
+	for i := range oldNAs {
+		oldNodeAffinity := oldNAs[i].(map[string]interface{})
+		newNodeAffinity := newNAs[i].(map[string]interface{})
+		if oldNodeAffinity["key"] != newNodeAffinity["key"] {
+			return true
+		}
+		if oldNodeAffinity["operator"] != newNodeAffinity["operator"] {
+			return true
+		}
+		if !reflect.DeepEqual(oldNodeAffinity["values"].(*schema.Set).List(), newNodeAffinity["values"].(*schema.Set).List()) {
```

Comment:
Does this compare without order? My understanding is that *schema.Set.List() is unstable, and that DeepEqual on an array will only return true if they are in the same order.

Reply:
Addressed this by using https://github.com/GoogleCloudPlatform/magic-modules/blob/885dc09cb3155e7586936725e5ebc329f6206697/mmv1/third_party/terraform/utils/utils.go#L222 which sorts the list